app/models/page/landing.rb

```diff
@@ -17,9 +18,12 @@ class Page::Landing < Page
   has_settings :layout_type
 
   validates :settings_layout_type, inclusion: { in: :settings_layout_type_enum }
+  validates :collection, presence: true, uniqueness: true
 
   delegate :settings_layout_type_enum, to: :class
 
+  before_create :set_slug if :collection
```

Comment:
Could you do this using an :if option to the callback instead of having the condition inline? Per http://guides.rubyonrails.org/v4.2/active_record_callbacks.html#conditional-callbacks